### ISSUE TITLE
Crash in ResultListFragment

### ISSUE THREAD

**Issue by caarmen** [caarmen/poet-assistant#81]
Fewer crashes, only Marshmallow:
```
Caused by: java.lang.NullPointerException: 
  at ca.rmen.android.poetassistant.main.dictionaries.ResultListFragment.query(ResultListFragment.java:142)
  at ca.rmen.android.poetassistant.main.dictionaries.search.Search.search(Search.java:89)
  at ca.rmen.android.poetassistant.main.MainActivity.onNewIntent(MainActivity.java:6196)
  at android.app.Instrumentation.callActivityOnNewIntent(Instrumentation.java:1219)
```
More crashes, mostly Marshmallow:
```
java.lang.NullPointerException: 
  at ca.rmen.android.poetassistant.main.dictionaries.ResultListFragment.query(ResultListFragment.java:142)
  at ca.rmen.android.poetassistant.main.dictionaries.search.Search.lambda$-ca_rmen_android_poetassistant_main_dictionaries_search_Search_lambda$1(Search.java:104)
  at ca.rmen.android.poetassistant.main.dictionaries.search.-$Lambda$32.$m$0(Unknown Source:0)
  at ca.rmen.android.poetassistant.main.dictionaries.search.-$Lambda$32.run(Unknown Source:0)
  at android.os.Handler.handleCallback(Handler.java:733)
```
More crashes, mostly 4.4:
```
java.lang.NullPointerException: 
  at ca.rmen.android.poetassistant.main.dictionaries.ResultListHeaderFragment.setHeader(ResultListHeaderFragment.java:86)
  at ca.rmen.android.poetassistant.main.dictionaries.ResultListFragment.query(ResultListFragment.java:142)
  at ca.rmen.android.poetassistant.main.dictionaries.search.Search.lambda$-ca_rmen_android_poetassistant_main_dictionaries_search_Search_lambda$1(Search.java:104)
  at ca.rmen.android.poetassistant.main.dictionaries.search.-$Lambda$32.$m$0(Unknown Source:0)
  at ca.rmen.android.poetassistant.main.dictionaries.search.-$Lambda$32.run(Unknown Source:0)
  at android.os.Handler.handleCallback(Handler.java:743)
```
Code:
```java
    public void query(String query) {
        Log.d(TAG, mTab + ": query() called with: " + "query = [" + query + "]");
        mHeaderFragment.setHeader(query);
        mHeaderFragment.setFilter(null);
```
`mHeaderFragment` is null or `ResultListHeaderFragment.mBinding` is null.

**Comment by caarmen** [comment]
Looks like this may happen when the app opens these deeplinks:
* poetassistant://rhymer/someword
* poetassistant://thesaurus/someword
* poetassistant://dictionary/someword

**Comment by caarmen** [comment]
Related to #19

**Comment by caarmen** [comment]
Still happens:
```
java.lang.NullPointerException: 
  at ca.rmen.android.poetassistant.main.dictionaries.ResultListFragment.query(ResultListFragment.java:145)
  at ca.rmen.android.poetassistant.main.dictionaries.search.Search.lambda$search$0(Search.java:87)
  at ca.rmen.android.poetassistant.main.dictionaries.search.Search$$Lambda$1.run(Unknown Source:0)
  at android.os.Handler.handleCallback(Handler.java:739)
  at android.os.Handler.dispatchMessage(Handler.java:95)
  at android.os.Looper.loop(Looper.java:148)
  at android.app.ActivityThread.main(ActivityThread.java:7325)
  at java.lang.reflect.Method.invoke(Native Method:0)
  at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
  at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
```

**Comment by caarmen** [comment]
Maybe fixed in 1.18.3...
